### ISSUE TITLE
Suggestions to speed up Anki OCR Workflow 

### ISSUE THREAD

**Issue by sephibox** [mvysny/aedict#764]
It's a great tool and I love it since it helps make my phone all I need to look up Kanji.

The biggest flaw for me is the impractical interface workflow/usability of the GUI when using it mainly as a Kanji lookup tool for Aedict with the inbuilt phone camera. 

It requires up to 11 button presses to get from Aedict to the OCR App and then having the results back in Aedict, when you want to use the phone camera to find look sth up. 

Since this is my main use for it, I wonder if the workflow could be simplified with a few configuration options.

To show what I mean, these are the steps currently necessary to look up a Kanji with the Camera option of Aedict OCR starting from Aedict.

1. (In Aedict if not on omnisearch screen) Press Menu Button 
2. Press Kanji Search Button (or Kanjipad Button when on Omnisearch screen)
3. Press Aedict OCR Button
4. (in OCR) Press Camera Mode Button
5. Press Camera Shutter Button to take Photo
6. Press Rotate Button, because camera orientation is ignored always defaulting to landscape mode. 
(I always take OCR pictures in Portrait orientation because of the vertical japanese texts and also I don't want to rotate my phone twice everytime I use it for this purpose)
7. Press mark button
8. Mark Kanji with selection box (very fiddly btw) or use finger selection
9. Press Checkmark button to confirm selection and show results
10. Press menu button
11. Press Analyze with Aedict.
-> Back in Aedict to lookup Kanji

This is very time consuming and feels clunky.

It would be a great improvement to the workflow if the following features could be implemented in some way:

1. Have a GUI tuning option in Aedict to add "OCR Camera Button" on top or bottom of Aedict main screen. This Button could directly launch Aedict OCR in Cameramode without having to go to Kanjipad. 
->1 Button Press and it's ready to take picture of Kanji

2. Press Button to take Photo, then directly open picture in marking mode, rotating the picture in the orientation the camera took it.
Alternatively: Have option to always show pictures in Portrait or Landscape orientation

3. Press Checkmark button to confirm selection and show results

4. Have "Analyze with Aedict" Button on results screen to directly get back to Aedict with the results.   

This would result in a much faster workflow. 

I don't know wether this is possible or maybe I am just using the App wrong? Any help would be appreciated. 

Thank you very much

**Comment by mvysny** [comment]
Thank you very much for your valuable input. Please be patient - I barely find the time to improve the Aedict tools. I know the UI design is far from perfect and would benefit from a complete overhaul, but there is simply no time :-(

However, it is true that I have not pushed any update to the OCR app in a long time, so I will try to fix this in a reasonable time frame.

**Comment by mvysny** [comment]
Yes, the selection box turned out to be a disaster :) Since I'm no UX expert here, I'll probably rip off the Google Translator - they have a really nice UI there.

**Comment by sephibox** [comment]
By all means take your time, I'm so thankful that you are still so active and always helpful and motivated to improve on your great work. I'd probably buy anything you make if I had not already done that. :-)

As for the selection box, I think the finger-selection tool works great imo! I don't even use anything else.

The main problem of the selection box controls is the rotation, it is very sensitive and rotates kind of unpredictably. Maybe having rotation deactivated by default could help, so you just draw a box parallel to the screen edges, since photos are usually already taken pretty straight.

**Comment by mvysny** [comment]
Thank you so much, I'm really flattered and happy that the Aedict suite is of use to you. Support is always appreciated since it's tough maintaining such a large project (especially since Aedict Online is around) on my free time (since the income is nowhere near to be able to sustain me, unfortunately). Anyways, you can take a look at https://www.patreon.com/aedict , maybe you'll like the idea behind that.

Yes, the rotation is highly annoying when the rectangle is very small - sorry about that ;) I'll think of something.
